samples/android-app-kotlin-minsdk-sample/src/main/java/com/squareup/wire/android/app/kotlin/minsdk/MainActivity.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.android.app.kotlin.minsdk
+
+import android.os.Bundle
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.adapter
+import com.squareup.wire.WireJsonAdapterFactory
+
+class MainActivity : AppCompatActivity() {
+
+  @OptIn(ExperimentalStdlibApi::class)
+  private val jsonAdapter = Moshi.Builder()
+    .add(WireJsonAdapterFactory())
+    .build()
+    .adapter<SomeText>()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+    val someText = checkNotNull(jsonAdapter.fromJson("""{"value": "Hi"}"""))
+    findViewById<TextView>(R.id.text_view).text = someText.value_
```

Comment:
Since we're fixing a runtime problem, we need this code to run on CI to be useful. We thus need a test that execute this path.

Reply:
hmmmm does that mean we need androidTests with an emulator...? hmmm

Reply:
Yup - I would love to write a test but I am both unfamiliar with the CI setup, and I don't think the current build configuration supports any of the ways I can think of to add a regression test for this, so your input would be helpful on what you think would be best. I don't want to introduce something unmaintainable just for this one test.

Right now I can think of a few ways this might be done, all of which are pretty icky in the context of what exists in the project already:
- Run unit tests on an older JDK: don't think the project has any support for this
- Use a mocking framework (e.g. MockK or Mockito): one is not already used in the project
- Use Robolectric: I don't even know if it would pick this up, but it is not part of this project
- Run instrumentation tests: Could use Gradle managed devices, or Firebase Test Lab. For the former, would need bare metal machines (which are probably already being used in the form of Mac agents?)

Let me know what you think.

Reply:
Just what I need, thanks! Will do.

By the way, it looks like `com.squareup.wire.GrpcClientTest > streamingRequestFailureWithUnexpectedHttpCode` might be [flaky](https://github.com/square/wire/actions/runs/8467516909/job/23198506065?pr=2881#step:6:1392)

Reply:
Added - fwiw I wasn't able to use assertj because it depends on other JDK APIs that can't be desugared